browser/ntp_background/android/ntp_background_images_bridge.cc

```diff
@@ -193,6 +193,42 @@ NTPBackgroundImagesBridge::CreateBrandedWallpaper(
       ConvertUTF8ToJavaString(env, wallpaper_id ? *wallpaper_id : ""));
 }
 
+base::android::ScopedJavaLocalRef<jobject>
+NTPBackgroundImagesBridge::CreateSponsoredRichMedia(
```

Comment:
I don't understand what is going on here. If this is functionality for desktop and android you should be exposing a single mojo interface to both the webui and java code as high level actions, but it looks like the java code has separate logic? 

Also in general we should be using mojo instead of jni bridge classes for android

Reply:
We need to pass `isRichMedia` boolean to the Java part because the logic in `BraveNewTabPageLayout` java class should be able to decide if we are showing native ThinWebView on Android NTP (which shows NewTabTakeover WebUI).

After NewTabTakeover WebUI is created and shown, it start communication via mojom.

Reply:
`NTPBackgroundImagesBridge` is a legacy class which uses JNI to do communication between Java and C++. 
Raised an refactoring issue for use `mojom` instead of `NTPBackgroundImagesBridge` in a follow-up PR: https://github.com/brave/brave-browser/issues/44211